x/gov/keeper/deposit.go

```diff
@@ -162,6 +163,73 @@ func (keeper Keeper) AddDeposit(ctx sdk.Context, proposalID uint64, depositorAdd
 	return activatedVotingPeriod, nil
 }
 
+// chargeDeposit will charge proposal cancellation fee (deposits * proposal_cancel_burn_rate)  and
+// send to a destAddress if defined or burn otherwise.
+// Remaining funds are send back to the depositor.
+func (keeper Keeper) chargeDeposit(ctx sdk.Context, proposalID uint64, destAddress, proposalCancelRate string, totalDeposits []sdk.Coin) error {
+	store := ctx.KVStore(keeper.storeKey)
+
+	rate := sdk.MustNewDecFromStr(proposalCancelRate)
+	var cancellationCharges sdk.Coins
+
+	if rate.IsPositive() {
+		for _, deposit := range totalDeposits {
+			burnAmount := sdk.NewCoin(
+				deposit.Denom,
+				sdk.NewDecFromInt(deposit.Amount).Mul(rate).RoundInt(),
+			)
+			cancellationCharges = cancellationCharges.Add(burnAmount)
+		}
+	}
+
+	// burn the cancellation fee or sent to cancellation changes to destination address
+	if !cancellationCharges.IsZero() {
+		// get the distribution module account address
+		distributionAddress := keeper.authKeeper.GetModuleAddress(disttypes.ModuleName)
+		if distributionAddress.String() == destAddress {
+			err := keeper.distrkeeper.FundCommunityPool(ctx, cancellationCharges, keeper.ModuleAccountAddress())
+			if err != nil {
+				return err
+			}
+		} else {
+			if len(destAddress) == 0 {
+				// burn the remaining deposits also
+				err := keeper.bankKeeper.BurnCoins(ctx, types.ModuleName, cancellationCharges)
+				if err != nil {
+					return err
+				}
+			} else {
+				destAccAddress := sdk.MustAccAddressFromBech32(destAddress)
+				err := keeper.bankKeeper.SendCoinsFromModuleToAccount(ctx, types.ModuleName, destAccAddress, cancellationCharges)
+				if err != nil {
+					return err
+				}
+			}
+		}
+	}
+
+	// send remaining (deposits - cancellationCharges) to depositors
+	for _, deposits := range keeper.GetDeposits(ctx, proposalID) {
```

Comment:
can you check the `totalDeposits` - `cancellationCharges` != 0, if not just ignore this operations may be?

Reply:
we can use 
```
// rate is not equal to 1 then we have ramaining deposits after the cancellation charges
rate := sdk.MustNewDecFromStr(proposalCancelRate)
if !rate.Equal(sdk.MustNewDecFromStr("1")) {
  // send remaining deposits to depositors 
}
```

instread `sub` between the coins